waspc/data/Generator/templates/server/src/dbClient.js

```diff
@@ -12,6 +13,41 @@ const createDbClient = () => {
   const prismaClient = new Prisma.PrismaClient()
 
   {=# isAuthEnabled =}
+
+  prismaClient.$use(async (params, next) => {
+    // Ensure strong plaintext password. Must come before hashing middleware.
+    // Throws an EntityValidationError on the first validation that fails.
+    if (params.model === '{= userEntityUpper =}' && params.action === 'create') {
+        const data = params.args.data || {}
+
+        const defaultValidations = [
+          { name: 'email must be present', fn: data => !!data.email },
+          { name: 'password must be present', fn: data => !!data.password },
+          { name: 'password must be at least 8 characters', fn: data => data.password.length >= 8 },
+          { name: 'password must contain a number', fn: data => /\d/.test(data.password) },
+        ]
+
+        let validations = params.args._waspSkipDefaultValidations ? [] : defaultValidations
+        if (Array.isArray(params.args._waspCustomValidations)) {
+          validations = validations.concat(params.args._waspCustomValidations)
+        }
+
+        for (const validation of validations) {
+          if (!validation.fn(data)) {
+            throw new EntityValidationError('{= userEntityUpper =}', validation.name)
+          }
+        }
+
+        // Remove from downstream Prisma processing to avoid "Unknown arg" error
+        delete params.args._waspSkipDefaultValidations
+        delete params.args._waspCustomValidations
+    }
+
+    const result = next(params)
+
+    return result
```

Comment:
Why not just 
```suggestion
    return next(params)
```

Reply:
Aha, I guess you copied that from below! Well, we can fix both :D (if it needs fixing).

Reply:
Haha, yeah copy paste, but I can fix in both! Thanks! 👍🏻

Reply:
Actually, you flagged a different, but actual problem with the original implementation. It needed to be `const result = await next(params)` instead: https://www.prisma.io/docs/concepts/components/prisma-client/middleware

Reply:
oops :D

Reply:
Aha! Well, looking at those examples, they way they split it like that was to give you an opportunity to observe what the middlewares (hah plural?) after your middleware returned. But we don't need that so we can just do `return next(params)`.

What `next(params)` returns is a Promise. using `await` on it effectively wraps it into another promise that manipulates the returned value of the first promise. But we don't need that so we can just return it.
All of these have the same effect:
```js
const resultPromise = next(params)
return resultPromise
```

```js
const result = await next(params)
return result
```

```js
return next(params)
```

Only in the second one though you can explore the results, if you insert some code between those lines.

So I would go with the simplest one, `return next(params)`, and if somebody needs to inspect the return value of `next(params)` before we forward it, they can use `await`.

Reply:
Ah, good point/explanation for the rusty like me @Martinsos and I should never doubt @matijaSos JS foo. Lesson learned, lesson learned. 😄

Reply:
Ha glad it was helpful, but please do always doubt it as we are certainly not JS experts :D!

Reply:
This has been addressed here: https://github.com/wasp-lang/wasp/pull/376/commits/3ffbc62c75cf4a43363b90fc8c2b1480309891ba Thanks!